src/Microsoft.AspNetCore.Authentication.OpenIdConnect/Events/OpenIdConnectEvents.cs

```diff
@@ -27,10 +28,17 @@ public class OpenIdConnectEvents : RemoteAuthenticationEvents
         public Func<MessageReceivedContext, Task> OnMessageReceived { get; set; } = context => Task.CompletedTask;
 
         /// <summary>
-        /// Invoked before redirecting to the identity provider to authenticate. This can be used to set ProtocolMessage.State
-        /// that will be persisted through the authentication process. The ProtocolMessage can also be used to add or customize
+        /// Invoked before redirecting to the identity provider to authenticate. This can be used to set
+        /// values in <c>ProtocolMessage.State</c> that will be persisted through the authentication process.
+        /// The <see cref="RedirectContext.ProtocolMessage"/> can also be used to add or alter
         /// parameters sent to the identity provider.
         /// </summary>
+        /// <remarks>
+        /// While this can be used to rewrite the redirect URLs if the app is behind a proxy server, its best to use
+        /// <c>ForwardedHeadersOptions</c>. Please see: https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/proxy-load-balancer?view=aspnetcore-2.1.
```

Comment:
We can't use direct links in doc comments, the links are unstable. If it were critical we'd use a forward link.